java/icicle/MemoryState.java

```diff
@@ -0,0 +1,186 @@
+import java.util.*;
+import java.time.LocalDate;
+
+class MemoryState<T> implements IcicleState<T>
```

Comment:
you should try writing bash that writes php (I still stand by my decision, it was more pleasant than writing php directly).